tsl/src/continuous_aggs/materialize.c

```diff
@@ -163,23 +171,33 @@ continuous_agg_materialize(int32 materialization_id, bool verbose)
 				 NameStr(cagg_data.user_view_name));
 	}
 
+	caggs = ts_continuous_aggs_find_by_raw_table_id(cagg_data.raw_hypertable_id);
+	drain_invalidation_log(cagg_data.raw_hypertable_id, &invalidations);
+	catalog = ts_catalog_get();
+	materialization_invalidation_log_table_relation =
+		heap_open(catalog_get_table_id(catalog, CONTINUOUS_AGGS_MATERIALIZATION_INVALIDATION_LOG),
+				  RowExclusiveLock);
+	insert_materialization_invalidation_logs(caggs,
+											 invalidations,
+											 materialization_invalidation_log_table_relation);
 	if (materializing_new_range)
 		invalidation_threshold_set(cagg_data.raw_hypertable_id, new_invalidation_threshold);
 
+	/* release lock on materialization_invalidation_log table */
+	heap_close(materialization_invalidation_log_table_relation, RowExclusiveLock);
```

Comment:
we shouldn't release the lock ourselves, `CommitTransactionCommand` will handle it

Reply:
modified to relation_close with NoLock.